ml-agents/mlagents/trainers/tests/test_stats.py

```diff
@@ -0,0 +1,76 @@
+import unittest.mock as mock
+import os
+
+from mlagents.trainers.stats import StatsReporter, TensorboardWriter
+
+
+def test_stat_reporter_add_summary_write():
+    # Test add_writer
+    StatsReporter.writers.clear()
+    mock_writer1 = mock.Mock()
+    mock_writer2 = mock.Mock()
+    StatsReporter.add_writer(mock_writer1)
+    StatsReporter.add_writer(mock_writer2)
+    assert len(StatsReporter.writers) == 2
+
+    # Test add_stats and summaries
+    statsreporter1 = StatsReporter("category1")
+    statsreporter2 = StatsReporter("category2")
+    for i in range(10):
+        statsreporter1.add_stat("key1", float(i))
+        statsreporter2.add_stat("key2", float(i))
+
+    statssummary1 = statsreporter1.get_stats_summaries("key1")
+    statssummary2 = statsreporter2.get_stats_summaries("key2")
+
+    assert statssummary1.num == 10
+    assert statssummary2.num == 10
+    assert statssummary1.mean == 4.5
+    assert statssummary2.mean == 4.5
+    assert round(statssummary1.std, 1) == 2.9
+    assert round(statssummary2.std, 1) == 2.9
+
+    # Test write_stats
+    step = 10
+    statsreporter1.write_stats(step)
+    mock_writer1.write_stats.assert_called_once_with("category1", "key1", 4.5, step)
+    mock_writer2.write_stats.assert_called_once_with("category1", "key1", 4.5, step)
+
+
+def test_stat_reporter_text():
+    # Test add_writer
+    mock_writer = mock.Mock()
+    StatsReporter.writers.clear()
+    StatsReporter.add_writer(mock_writer)
+    assert len(StatsReporter.writers) == 1
+
+    statsreporter1 = StatsReporter("category1")
+
+    # Test write_text
+    step = 10
+    statsreporter1.write_text("this is a text", step)
+    mock_writer.write_text.assert_called_once_with("category1", "this is a text", step)
+
+
+@mock.patch("mlagents.tf_utils.tf.Summary")
+@mock.patch("mlagents.tf_utils.tf.summary.FileWriter")
+def test_tensorboard_writer(mock_filewriter, mock_summary):
+    # Test write_stats
+    base_dir = "base_dir"
+    category = "category1"
+    tb_writer = TensorboardWriter(base_dir)
```

Comment:
Will this actually make a directory somewhere if I run tests locally? If so, you should do something like
``` python
import tempfile
...
with tempfile.TemporaryDirectory(prefix="unittest-") as base_dir:
    # rest of test here
```
Then the directory will get automatically cleaned up when the context manager closes.

Reply:
Confirmed that this create `./base_dir` and `./base_dir/category1` locally. Please use a tempfile.

Reply:
Test now uses tempfile.